src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java

```
@@ -0,0 +1,236 @@
+                if (StringUtils.isNotEmpty(localActive)) {
+                    if (Boolean.valueOf(localActive.trim())) {
+                        ftp.enterLocalActiveMode();
+                    } else {
+                        ftp.enterLocalPassiveMode();
+                    }
+                } else {
+                    ftp.enterLocalPassiveMode();
+                }
+                inputStream = new ByteArrayInputStream(messageContext.getEnvelope().getBody().getFirstElement().
+                        toString().getBytes());
+                if (StringUtils.isNotEmpty(targetPath)) {
+                    ftp.changeWorkingDirectory(targetPath);
+                    ftp.storeFile(targetFile, inputStream);
+                    if (log.isDebugEnabled()) {
+                        log.debug("Successfully FTP transfered the File");
+                    }
+                }
+                // check that control connection is working
+                if (log.isDebugEnabled()) {
+                    log.debug("The code received from the server." + ftp.noop());
+                }
+                resultStatus = true;
+            } else {
+                ftp.logout();
+                handleException("Error while login ftp server.", messageContext);
+            }
+        } catch (FTPConnectionClosedException e) {
+            // log.error("Server closed connection " + e.getMessage(), e);
+            handleException("Server closed connection: " + e.getMessage(), e, messageContext);
+        } catch (IOException e) {
+            //log.error("Error occurred while uploading:" + e.getMessage(), e);
+            handleException("Could not connect to FTP ftpServer: " + e.getMessage(), e, messageContext);
+        } finally {
+            if (ftp.isConnected()) {
+                try {
+                    ftp.disconnect();
+                    ftp.logout();
+                } catch (IOException f) {
+                    // do nothing
```

This should not be silently ignored IMO. Could you add a log ?

---

Fixed in wso2-extensions/esb-connector-file#28